frontend/src/metabase/parameters/utils/dashboards.js

```diff
@@ -63,8 +63,8 @@ export function isDashboardParameterWithoutMapping(parameter, dashboard) {
   return parameterExistsOnDashboard && !parameterHasMapping;
 }
 
-function getMappingTargetField(card, mapping, metadata) {
-  if (!card) {
+export function getMappingTargetField(card, mapping, metadata) {
+  if (!card || !card.dataset_query) {
```

Comment:
nit: could be `if (!card?.dataset_query)`

Reply:
yep, updated in https://github.com/metabase/metabase/pull/20678/commits/ca8ef992d01559d963496684db80c2d122dbc323